cloudevents/extensions/recordedtime.md

```diff
@@ -0,0 +1,106 @@
+# Recorded Time Extension
+
+This extension defines an attribute that represents the time when an
+[_occurrence_](https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#occurrence)
+was recorded in a particular
+[_event_](https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#event),
+which is the time when the CloudEvent was created by a producer.
+
+This attribute is distinct from the [`time`
+attribute](https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#time),
+which, according to the CloudEvents specification, SHOULD be the time when the
+occurrence happened, if it can be determined.
+
+This attribute makes it possible to represent
+[bitemporal](https://en.wikipedia.org/wiki/Bitemporal_modeling) data with
+CloudEvents so that, for every event, both of the following times can be known
+by consumers:
+
+- _Occurrence time_: timestamp of when the occurrence recorded in the event
+  happened, which corresponds to the `time` attribute.
+- _Recorded time_: the timestamp of when the occurrence was recorded in a
+  specific CloudEvent instance, which is represented by this extension.
+
+## Attributes
+
+### recordedtime
+
+- Type: `Timestamp`
+- Description: Timestamp of when the occurrence was recorded in this CloudEvent,
+  i.e. when the CloudEvent was created by a producer. If this attribute is used
```

Comment:
I think the 2nd sentence is unneeded and could be misleading because it implies that if `recordedtime` isn't used then `time` might mean something different. Or are you just trying to say: "If this attribute is used then `time` SHOULD be present in the CloudEvent"  ?? I think that would be ok.

Reply:
The CloudEvent spec says:

> If the time of the occurrence cannot be determined then this attribute MAY be set to some other time (such as the current time) by the CloudEvents producer

The second sentence here is an attempt to negate that. However, it is probably best to omit it, it really does not add much value and could add negative value.

Reply:
I removed this now.

Reply:
Removed it now.